internal/controllers/nmc_reconciler.go

```diff
@@ -555,104 +565,138 @@ func (h *nmcReconcilerHelperImpl) SyncStatus(ctx context.Context, nmcObj *kmmv1b
 	return errors.Join(errs...)
 }
 
-func (h *nmcReconcilerHelperImpl) UpdateNodeLabelsAndRecordEvents(ctx context.Context, nmc *kmmv1beta1.NodeModulesConfig) error {
-	node := v1.Node{}
-	if err := h.client.Get(ctx, types.NamespacedName{Name: nmc.Name}, &node); err != nil {
-		return fmt.Errorf("could not get node %s: %v", nmc.Name, err)
-	}
+type labelPreparationHelper interface {
+	getDeprecatedKernelModuleReadyLabels(node v1.Node) sets.Set[string]
+	getNodeKernelModuleReadyLabels(node v1.Node) sets.Set[types.NamespacedName]
+	getSpecLabelsAndTheirConfigs(nmc *kmmv1beta1.NodeModulesConfig) map[types.NamespacedName]kmmv1beta1.ModuleConfig
+	getStatusLabelsAndTheirConfigs(nmc *kmmv1beta1.NodeModulesConfig) map[types.NamespacedName]kmmv1beta1.ModuleConfig
+}
+type labelPreparationHelperImpl struct{}
 
-	// get all the kernel module ready labels of the node
+func NewLabelPreparationHelper() labelPreparationHelper {
```

Comment:
We don't need this function to be public. This is an inner interface to the package, it can be initialized by the NewNMCReconcile , just like all other helper interfaces

Reply:
I agree, but you will need to manually create the object when testing in order to use a mock since there is no dependency injection anymore.

Reply:
we do it all the time in the unit-test

Reply:
done.
now it is initialized in `newNMCReconcilerHelper` instead of `NewNMCReconcile` because we need it in  `nmcReconcilerHelperImpl` struct